source/common/tcp_proxy/upstream.cc

```diff
@@ -258,6 +354,11 @@ void HttpConnPool::onPoolFailure(ConnectionPool::PoolFailureReason reason,
   callbacks_->onGenericPoolFailure(reason, failure_reason, host);
 }
 
+void HttpConnPool::onUpstreamHostSelected(Upstream::HostDescriptionConstSharedPtr host) {
+  upstream_->setConnPoolCallbacks(std::make_unique<HttpConnPool::Callbacks>(
+      *this, host, downstream_info_.downstreamAddressProvider().sslConnection()));
+}
+
 void HttpConnPool::onPoolReady(Http::RequestEncoder& request_encoder,
```

Comment:
small question - when using Router::UpstreamRequest is ``HttpConnPool::onPoolReady`` going to be called? or just the ``HttpConnPool::onUpstreamHostSelected``?

Reply:
just the HttpConnPool::onUpstreamHostSelected